src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutSubItem.cs

```diff
@@ -1,47 +1,882 @@
+
+			spParentPresenter = GetParentMenuFlyoutPresenter();
+			if (spParentPresenter != null)
+			{
+				IMenuPresenter openedSubPresenter;
+
+				openedSubPresenter = (spParentPresenter as IMenuPresenter).SubPresenter;
+				if (openedSubPresenter != null)
+				{
+					ISubMenuOwner subMenuOwner;
+
+					subMenuOwner = openedSubPresenter.Owner;
+					if (subMenuOwner != null && subMenuOwner != this)
+					{
+						openedSubPresenter.CloseSubMenu();
+					}
+				}
+			}
+
+
+		}
+
+		bool IsOpen => m_tpPopup?.IsOpen ?? false;
+
+		Control
+		CreateSubPresenter()
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: CreateSubPresenter.", this));
+#endif // MFSI_DEBUG
+
+			MenuFlyoutPresenter spPresenter = new MenuFlyoutPresenter();
+
+			// Specify the sub MenuFlyoutPresenter
+			(spPresenter as MenuFlyoutPresenter).IsSubPresenter = true;
+			(spPresenter as IMenuPresenter).Owner = this;
+
+			return spPresenter;
+		}
+
+		void UpdateParentOwner(MenuFlyoutPresenter parentMenuFlyoutPresenter)
+		{
+			MenuFlyoutPresenter parentPresenter = parentMenuFlyoutPresenter;
+			if (parentPresenter == null)
+			{
+				parentPresenter = GetParentMenuFlyoutPresenter();
+			}
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: UpdateParentOwner - parentPresenter=0x%p.", this, parentPresenter));
+#endif // MFSI_DEBUG
+
+			if (parentPresenter != null)
+			{
+				ISubMenuOwner parentSubMenuOwner;
+				parentSubMenuOwner = (parentPresenter as IMenuPresenter).Owner;
+
+#if MFSI_DEBUG
+				IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: UpdateParentOwner - parentSubMenuOwner=0x%p.", this, parentSubMenuOwner));
+#endif // MFSI_DEBUG
+
+				if (parentSubMenuOwner != null)
+				{
+					((ISubMenuOwner)this).ParentOwner = parentSubMenuOwner;
+				}
+			}
+		}
+
+		// Set the popup open or close status for MenuFlyoutSubItem and ensure the
+		// focus to the current presenter.
+		void SetIsOpen(bool isOpen)
+		{
+			bool isOpened = false;
+
+			isOpened = m_tpPopup.IsOpen;
+
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: SetIsOpen isOpen=%d, isOpened=%d.", this, isOpen, isOpened));
+#endif // MFSI_DEBUG
+
+			if (isOpen != isOpened)
+			{
+				(m_tpPresenter as IMenuPresenter).Owner = isOpen ? this : null;
+
+				MenuFlyoutPresenter parentPresenter;
+				parentPresenter = GetParentMenuFlyoutPresenter();
+
+				if (parentPresenter != null)
+				{
+					(parentPresenter as IMenuPresenter).SubPresenter = isOpen ? m_tpPresenter as MenuFlyoutPresenter : null;
+
+					IMenu owningMenu;
+					owningMenu = (parentPresenter as IMenuPresenter).OwningMenu;
+
+					if (owningMenu != null)
+					{
+						(m_tpPresenter as IMenuPresenter).OwningMenu = isOpen ? owningMenu : null;
+					}
+
+					UpdateParentOwner(parentPresenter);
+				}
+
+				// UNO TODO
+				//VisualTree* visualTree = VisualTree.GetForElementNoRef(GetHandle());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)